airbyte-cdk/python/airbyte_cdk/sources/streams/core.py

```diff
@@ -384,6 +385,14 @@ def get_updated_state(
         """
         return {}
 
+    def get_cursor(self) -> Optional[Cursor]:
+        """
+        A Cursor is an interface that a stream can implement to manage how its internal state is read and updated while
+        reading records. Historically, Python connectors had no concept of a cursor to manage state. Python streams need
+        need to define a cursor implementation and override this method to manage state through a Cursor.
+        """
+        return None
+
     def _get_checkpoint_reader(
```

Comment:
yeah i'm not sure. I did fiddle around with it a bit, but it's kinda hard to streamline it because the cursor reader needs the checkpoint reader. It is pretty hard to follow, but I think that's just a hazard of us having so many sync modes and flavors of CDK. :/ i might leave it as is for now since I don't have a good way to minimize the complexity and I tried to just keep it isolated with the `_get_checkpoint_reader()` function

Reply:
Fair enough, let's re-look at it when we've narrowed it down a bit 👍🏻